### PR TITLE
feat: improved webhook tracing

### PR DESCRIPTION
**Pull request by alnr**
https://github.com/ory/x/pull/761

### CODE REVIEW

File: driver/registry_default.go [hunk 1]
```diff
@@ -815,16 +816,13 @@ func (m *RegistryDefault) PrometheusManager() *prometheus.MetricsManager {
...
 		httpx.ResilientClientWithLogger(m.Logger()),
 		httpx.ResilientClientWithMaxRetry(2),
-		httpx.ResilientClientWithConnectionTimeout(30*time.Second))
-
-	tracer := m.Tracer(ctx)
-	if tracer.IsLoaded() {
-		opts = append(opts, httpx.ResilientClientWithTracer(tracer.Tracer()))
-	}
+		httpx.ResilientClientWithConnectionTimeout(30*time.Second),
+		httpx.ResilientClientWithTracer(noop.NewTracerProvider().Tracer("Ory Kratos")), // will use the tracer from a context if available
```
Comment (zepatrik) [inline driver/registry_default.go:824]: How can this use the tracer form the context if the context is not even passed to it?

File: selfservice/hook/web_hook.go [hunk 2]
```diff
@@ -525,12 +525,16 @@ func instrumentHTTPClientForEvents(ctx context.Context, httpClient *retryablehtt
...
 	}
 	httpClient.ResponseLogHook = func(_ retryablehttp.Logger, res *http.Response) {
-		res.Body = io.NopCloser(io.LimitReader(res.Body, 5<<20)) // read at most 5 MB from the response
-		resBody, _ := httputil.DumpResponse(res, true)
-		resBody = resBody[:min(len(resBody), 2<<10)] // truncate response body to 2 kB for event
+		// res.Body = io.NopCloser(io.LimitReader(res.Body, 5<<20)) // read at most 5 MB from the response
+		// resBody, _ := httputil.DumpResponse(res, true)
+		// resBody = resBody[:min(len(resBody), 2<<10)] // truncate response body to 2 kB for event
+		// TODO(@alnr): redact sensitive data
+		resBody := []byte("<redacted>")
```
Comment (zepatrik) [inline selfservice/hook/web_hook.go:537]: Todo?

File: driver/registry_default.go (hunk 1 above)
Comment (alnr) [reply driver/registry_default.go:824]: The tracer is retrieved from the request context:
```
req, err := http.NewRequest(ctx /* HERE */, GET, "http://localhost", nil)
```

File: selfservice/hook/web_hook.go (hunk 2 above)
Comment (alnr) [reply selfservice/hook/web_hook.go:537]: I haven't figured out how to do this cleanly. That's for another PR.

File: driver/registry_default.go (hunk 1 above)
Comment (zepatrik) [reply driver/registry_default.go:824]: Ah so here we instantiate the client, and the tracer does not matter. When the request is made, a context with the correct tracer is passed.
Correct?

Comment (alnr) [reply driver/registry_default.go:824]: Precisely. The `otelhttp` instrumentation checks for an existing span in the context. If it finds one, it uses the same tracer to start its own spans. If it doesn't find one, it uses the one passed here. Finally, it falls back to the global tracer (which we don't typicall configure).

**Review by zepatrik — APPROVED** [review]

**Pull request merged**